cmake/CheckCPUArchitecture.cmake

```diff
@@ -32,6 +32,8 @@ function(check_cpu_architecture ARCH VARIABLE)
     _internal_check_cpu_architecture("defined(__arm__) || defined(_M_ARM)" arm32 ${VARIABLE})
   elseif(ARCH STREQUAL "arm64")
     _internal_check_cpu_architecture("defined(__aarch64__) || defined(_M_ARM64)" arm64 ${VARIABLE})
+  elseif(ARCH STREQUAL "loongarch64")
+    _internal_check_cpu_architecture("defined(__loongarch64)" loongarch64 ${VARIABLE})

```

Comment:
@HecaiYuan Is this a correct way to detect loongarch64?
I got it [from here](https://github.com/gcc-mirror/gcc/blob/0c1b0a23f1fe7db6a2e391b7cb78cff900377772/gcc/config/loongarch/loongarch-c.cc#L71).

Reply:
No problem.